swap/cheque.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package swap
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// Encode encodes the cheque in the format used in the signing procedure
```

Comment:
@ralph-pichler tagging you since I think your input is relevant here.

I think for this we only need to change the signature to:
`MarshalBinary() (data []byte, err error)`

However, in this function I don't see an `error` variable. 

Can this encoding ever fail?

Reply:
I don't think it can fail anywhere. We could just always return `nil` as error.

Reply:
fixed